grails-app/domain/org/pih/warehouse/order/OrderItem.groovy

```diff
@@ -350,7 +350,7 @@ class OrderItem implements Serializable, Comparable<OrderItem> {
     }
 
     def getInvoices() {
-        return invoiceItems*.invoice.unique()
+        return allInvoiceItems*.invoice.unique()
```

Comment:
@awalkowiak it's probably due to the fact, you don't need to join the `order_item` table anymore (as you've added the association), not the HQL itself.
Still I'm 99% sure that a similar query is invoked when you call the `getAllInvoiceItems()`: (pseudocode)
```sql
1. select shipmentItem from order_shipment inner join shipment_item on shipment_item.id = order_shipment.shipment_item_id
2. select invoiceItem from shipment_invoice inner join invoice_item on invoice_item.id = shipment_invoice.invoice_item_id
3. select invoiceItem from order_invoice inner join invoice_item on invoice_item.id = order_invoice.invoice_item_id
```
so those 3 queries would be run, it's a blind guess, but am 99% that this happens at the moment, so there might still be a room to improvement if we run into any problems.

Reply:
> I think the similar thing happens for the allInvoiceItems - the difference is that there we actually might do more queries:

@kchelstowski Yeah I was thinking the same thing. It's essentially the same but possibly compounded.

@awalkowiak Agreed. I forgot that this was the refactoring. We don't need to deal with this now. I just want to be proactive on identifying any potential performance bottlenecks so we can be ready to address them. Again, i'm not worried if this method is invoked once within a request for a single order or invoice. But if we're unwittingly calling these methods inside loops and those invocations are generating lots of queries ... then we should be aware of that and make sure we're taking steps to avoid it. 

Editorial Note: My affinity for using transient properties (this "invoices" property) and overriding getter methods with these expensive getter method implementations is bordering on "code smell" as it gets us into trouble if we don't identify and understand the implications. I still prefer to use these transient getters because it makes the code cleaner, but it would be nice for the IDE to warn us when we're about to do something stupid.